day2.py

```diff
@@ -0,0 +1,111 @@
+from gophishUtility import *
+from gophish.models import *
+from math import ceil
+import os
+
+# This client has all methods attached to it necessary to create, read, and run our campaigns.
+gophishClient = GophishUtility.CreateGophishClient()
+
+# Gets all campaigns that are currently In Progress and are named something that can be found in the GophishUtility.groupNames list
+# in the gophishUtility.py file.
+def getOurCampaigns():
+    campaigns = [campaign for campaign in gophishClient.campaigns.get() if campaign.name in GophishUtility.groupNames and campaign.status != "Completed"]
+    return campaigns
+
+# Write Campaign Results out to the file that you name in the parameter
+def writeResultsTo(filename):
+    with open(filename, "w") as resultsFile:
+        resultsFile.write("First Name, Last Name, Email, Status, Group\n")
+        for campaign in campaigns:
+            for result in campaign.results:
+                resultsFile.write("{},{},{},{},{}\n".format(result.first_name, result.last_name, result.email, result.status, campaign.name))
+
+        resultsFile.close()
+
+# Sets up new groups and moves all persons who clicked the link into our control group for training.
+# Then it deletes all the groups and then recreates them with their new audience.
+def setUpNewGroups(campaigns):
+    groups = []
+    count = 0
+    for i in range(20):
+        newGroup = Group(name = GophishUtility.groupNames[i], targets=[])
+        groups.append(newGroup)
+
+    for campaign in campaigns:
+        for result in campaign.results:
+            newUser = User(first_name = result.first_name, last_name = result.last_name, email = result.email)
+            groupName = campaign.name if result.status != "Clicked Link" else campaign.name[:14]+"Control"
```

Comment:
This campaign.name should be changed into something more programmatic that figures out what should be used instead of relying on a slice always being right.